src/_components/AddAsset/AddAssetContainer.jsx

```diff
@@ -0,0 +1,179 @@
+    if(data.name === 'asset-category') {
+      this.setState({
+        filteredSubCategories:
+        filterSubCategories(this.props.subcategories, data.value)
+      });
+    } else if (data.name === 'asset-subcategory') {
+      this.setState({
+        filteredAssetTypes:
+        filterAssetTypes(this.props.assetTypes, data.value)
+      });
+    } else if (data.name === 'asset-types') {
+      this.setState({
+        filteredAssetMakes:
+        filterAssetMakes(this.props.assetMakes, data.value)
+      });
+    } else if (data.name === 'asset-makes') {
+      this.setState({
+        filteredModelNumbers:
+        filterModelNumbers(this.props.modelNumbers, data.value)
+      });
+    }
+  };
+
+  onSelectModelNumber = (event, data) => {
+    this.setState({ modelNumber: data.value });
+  };
+
+  onAddSerialNumber = (event) => {
+    this.setState({ serialNumber: event.target.value });
+  };
+
+  onAddAssetTag = (event) => {
+    this.setState({ assetTag: event.target.value });
+  };
+
+  onCreateAsset = (event, data) => {
+    this.props.createAsset({
+      "asset_code": this.state.assetTag,
+      "serial_number": this.state.serialNumber,
+      "model_number": this.state.modelNumber
+    });
+    event.target.reset();
+  };
+
+  render(){
+    return (
+      <AddAssetComponent {...this.props}
+        handleDropdownChanges={this.handleDropdownChanges}
+        onSelectModelNumber={this.onSelectModelNumber}
+        onAddSerialNumber={this.onAddSerialNumber}
+        onAddAssetTag={this.onAddAssetTag}
+        onCreateAsset={this.onCreateAsset}
+        filteredSubCategories={this.state.filteredSubCategories}
+        filteredAssetTypes={this.state.filteredAssetTypes}
+        filteredAssetMakes={this.state.filteredAssetMakes}
+        filteredModelNumbers={this.state.filteredModelNumbers}
+        modelNumber={this.state.modelNumber}
+        serialNumber={this.state.serialNumber}
+        assetTag={this.state.assetTag}
+      />
+    );
+  }
+}
+
+AddAssetComponent.propTypes = {
+  categoriesList: PropTypes.array,
+  subcategoriesList: PropTypes.array,
+  assetTypesList: PropTypes.array,
+  assetMakesList: PropTypes.array,
+  modelNumbersList: PropTypes.array,
+  assetsList: PropTypes.array,
+  loadCategories: PropTypes.func,
+  loadSubCategories: PropTypes.func,
+  loadAssetTypes: PropTypes.func,
+  loadAssetMakes: PropTypes.func,
+  loadModelNumbers: PropTypes.func,
+  createAsset: PropTypes.func
+};
+
+const mapStateToProps = (
+  { categoriesList,
+    subcategoriesList,
+    assetTypeList,
+    assetMakesList,
+    modelNumbersList,
+    assetsList
+   }) => {
+  const categories = categoriesList;
+  const subcategories = subcategoriesList;
+  const assetTypes = assetTypeList;
+  const assetMakes = assetMakesList;
+  const modelNumbers = modelNumbersList;
+  const assets = assetsList;
+  return  { categories, subcategories, assetTypes, assetMakes, modelNumbers, assets };
+};
+
+export default connect(mapStateToProps,
+  {
```

Comment:
Same as the previous comment, the opening brace should move to the previous line. I.e.
```
export default connect(mapStateToProps, {
  loadCategories,
  loadSubCategories,
  loadAssetTypes,
  loadAssetMakes,
  loadModelNumbers,
  createAsset
})(AddAssetContainer);
```